velox/common/memory/Memory.h

```diff
@@ -266,17 +274,84 @@ class ScopedMemoryPool final : public MemoryPool {
 // node tree.
 class MemoryAllocator {
  public:
-  // TODO: move to factory pattern with type trait.
   static std::shared_ptr<MemoryAllocator> createDefaultAllocator();
 
-  void* alloc(int64_t size);
-  void* allocZeroFilled(int64_t numMembers, int64_t sizeEach);
+  virtual ~MemoryAllocator() {}
+
+  virtual void* FOLLY_NULLABLE alloc(int64_t size);
+  virtual void* FOLLY_NULLABLE
+  allocZeroFilled(int64_t numMembers, int64_t sizeEach);
   // TODO: might be able to collapse this with templated class
-  void* allocAligned(uint16_t alignment, int64_t size);
-  void* realloc(void* p, int64_t size, int64_t newSize);
-  void*
-  reallocAligned(void* p, uint16_t alignment, int64_t size, int64_t newSize);
-  void free(void* p, int64_t size);
+  virtual void* FOLLY_NULLABLE allocAligned(uint16_t alignment, int64_t size);
+  virtual void* FOLLY_NULLABLE
+  realloc(void* FOLLY_NULLABLE p, int64_t size, int64_t newSize);
+  virtual void* FOLLY_NULLABLE reallocAligned(
+      void* FOLLY_NULLABLE p,
+      uint16_t alignment,
+      int64_t size,
+      int64_t newSize);
+  virtual void free(void* FOLLY_NULLABLE p, int64_t size);
+};
+
+// An allocator that uses memory::MappedMemory to allocate memory. We leverage
+// MappedMemory for relatively small allocations. Allocations less than 3/4 of
+// smallest size class and larger than largest size class are delegated to
+// jemalloc still.
+class MmapMemoryAllocator : public MemoryAllocator {
```

Comment:
Is this already implemented in #1434?

Reply:
Sorry that I have not seen this and responded in time. To answer your question. They are two different things. The implemented allocator in #1434 is an allocator in MappedMemory. This allocator though is going to be used by MemoryPool, which will delegate the job to MappedMemory by using this allocator.